### PR TITLE
feat: Add support for open-sourced multi-modal model

### PR DESCRIPTION
**Pull request by Appointat**
https://github.com/camel-ai/camel/issues/705

We refer to the integration of `InternLM-XComposer-2.5` model.

### CODE REVIEW

**Comment by Appointat** [comment]
This pr can be reviewed. I have integrate the capacity of the images for the opensource model. (the video part can be updated after this issue resolved https://github.com/camel-ai/camel/issues/806)

File: camel/messages/base.py [hunk 1]
```diff
@@ -373,6 +391,108 @@ def to_openai_user_message(self) -> OpenAIUserMessage:
...
+        Returns:
+            OpenAIUserMessage: The converted :obj:`OpenAIUserMessage` object.
+        """
+        hybird_content: List[Any] = []
+        hybird_content.append(
+            {
+                "type": "text",
+                "text": self.content,
+            }
+        )
```
Comment (WHALEEYE) [inline camel/messages/base.py:406]:
```suggestion
        hybird_content: List[Any] = [{
                "type": "text",
                "text": self.content,
            }]
```

Also there is a small typo in the name: `hybird -> hybrid`

**Review by WHALEEYE — APPROVED** [review]
Thanks! Except a small typo the PR LGTM.

File: camel/models/interlm_xcomposer_model.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    def token_counter(self) -> BaseTokenCounter:
+        r"""Initialize the token counter for the model backend.
+
+        Returns:
+            BaseTokenCounter: The token counter following the model's
+                tokenization style.
+        """
+        if not self._token_counter:
+            self._token_counter = OpenAITokenCounter(ModelType.GPT_3_5_TURBO)
+        return self._token_counter
```
Comment (lightaime) [inline camel/models/interlm_xcomposer_model.py:141]: Why we use `OpenAITokenCounter(ModelType.GPT_3_5_TURBO)` for multi-modal models?

File: examples/models/interlm_xcomposer_model_example.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+video_url = 'https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerBlazes.mp4'
+video_bytes = load_video_from_url(video_url)
+video_bytes_list = [video_bytes]
+image_url = 'https://upload.wikimedia.org/wikipedia/commons/thumb/d/dd/Gfp-wisconsin-madison-the-nature-boardwalk.jpg/2560px-Gfp-wisconsin-madison-the-nature-boardwalk.jpg'
+image = load_image_from_url(image_url)
+images = [image, image]
+
+
+ollama_model = ModelFactory.create(
+    model_platform=ModelPlatformType.INTERNLM,
```
Comment (lightaime) [inline examples/models/interlm_xcomposer_model_example.py:67]: It is loaded from huggingface. Why it is `ModelPlatformType.INTERNLM`?

**Comment by lightaime** [comment]
Thanks @Appointat and @WHALEEYE. But it seems this PR is not ready to merge and need significant revision.

File: camel/messages/base.py [hunk 4]
```diff
@@ -98,24 +104,26 @@ def make_assistant_message(
...
         ] = OpenAIVisionDetailType.AUTO,
         video_detail: Union[
             OpenAIVisionDetailType, str
         ] = OpenAIVisionDetailType.LOW,
+        video_type: Optional[str] = None,
     ) -> "BaseMessage":
         return cls(
             role_name,
-            RoleType.ASSISTANT,
+            RoleType.USER,
```
[the remark this answers is not in the document]
Comment (Appointat) [reply camel/messages/base.py:119]: Fixed, thanks.

**Pull request merged**